internal/osutils/autostart/autostart.go

```diff
@@ -1,50 +1,72 @@
 package autostart
 
 import (
+	"github.com/ActiveState/cli/internal/constants"
 	"github.com/ActiveState/cli/internal/errs"
 )
 
 const ConfigKeyDisabled = "SystrayAutoStartDisabled"
 
+type AppName string
+
+func (a AppName) String() string {
+	return string(a)
+}
+
+const (
+	Tray    AppName = constants.TrayAppName
+	Service         = constants.SvcAppName
+)
+
 type App struct {
-	Name string
-	Exec string
-	cfg  Configurable
+	Name    string
+	Exec    string
+	Args    []string
+	cfg     Configurable
+	options options
 }
 
 type Configurable interface {
 	Set(string, interface{}) error
 	IsSet(string) bool
 }
 
-func New(name, exec string, cfg Configurable) *App {
+func New(name AppName, exec string, args []string, cfg Configurable) *App {
 	return &App{
-		Name: name,
-		Exec: exec,
-		cfg:  cfg,
+		Name:    name.String(),
+		Exec:    exec,
+		Args:    args,
+		cfg:     cfg,
+		options: data[name],
```

Comment:
It's possible this could fail by programmer error. Do we want to handle this properly? Do we have a way of handling this?

Reply:
I've made it so that the only way you can get an `app` object is via the `New` function. I think this should cover it. Is there another way you can see?